scripts/rearchive-all-docs.js

```diff
@@ -61,20 +61,32 @@ async function rearchiveAllDocs() {
   // start from an objectId and run in ascending order, so we can resume later
   const query = {}
   const startId = params._[0]
+  const endId = params.e
   if (startId) {
-    const validator = new RegExp('^[0-9a-fA-F]{24}$')
-    if (!validator.test(startId)) {
-      console.error('Invalid object id')
-      return
+    if (!new RegExp('^[0-9a-fA-F]{24}$').test(startId)) {
+      throw new Error('Invalid start object id')
     }
     query._id = {
-      $gt: ObjectId(startId)
+      $gte: ObjectId(startId)
     }
     console.log(`Starting from object ID ${startId}`)
   } else {
     console.log('No object id specified. Starting from the beginning.')
   }
 
+  if (endId) {
+    if (!new RegExp('^[0-9a-fA-F]{24}$').test(endId)) {
+      throw new Error('Invalid end object id')
+    }
+    query._id = {
+      ...(query._id || {}),
+      ...{
+        $lte: ObjectId(endId)
+      }
+    }
```

Comment:
I had to look more than twice at this. WDYT about simplify it?
```suggestion
    query._id = query._id || {}
    query._id.$lte = ObjectId(endId)
```

<details>

```js
> query={}
{}
> query._id = query._id || {}
{}
> query._id.$lte = 'ObjectId(endId)'
'ObjectId(endId)'
> query
{ _id: { '$lte': 'ObjectId(endId)' } }
> query={}
{}
> query._id = {$gte: 'ObjectId(startId)'}
{ '$gte': 'ObjectId(startId)' }
> query._id = query._id || {}
{ '$gte': 'ObjectId(startId)' }
> query._id.$lte = 'ObjectId(endId)'
'ObjectId(endId)'
> query
{ _id: { '$gte': 'ObjectId(startId)', '$lte': 'ObjectId(endId)' } }
```
</details>

Reply:
Booriiing. :wink: 

You're right - what started out as a way to avoid assigning twice ended up sprawling. :+1: